source/webhooks/index.html.md.erb

```diff
@@ -0,0 +1,180 @@
+* a payment is refunded
+
+You can create webhooks in the GOV.UK Pay admin tool. After selected payment events, your webhook sends a webhook message using a `POST` request to a secure callback URL you’ve configured. Your service acknowledges receipt of the webhook message.
+
+## Set up a callback URL to receive a webhook message
+
+You need to set up an endpoint (callback URL) that can receive webhook messages before creating the webhook.
+
+You must set up your endpoint to receive the `POST` body from your webhook and respond with a `2xx` successful reply within 15 seconds.
+
+Your callback URL must use HTTPS and present a valid certificate. You can use a HTTP URL for test accounts. GOV.UK webhooks support TLS version 1.2.
+
+## Create a webhook
+
+You can create a webhook for your service from the [GOV.UK Pay admin tool](https://selfservice.payments.service.gov.uk/login).
+
+1. Select your service from the **Overview** page in the GOV.UK Pay admin tool.
+
+1. Select **Settings**.
+
+1. Select **Webhooks**.
+
+1. Select **Create a new webhook**.
+
+1. Enter a **Callback URL** and **Description**.
+
+1. Select the payment events that the webhook will send.
+
+1. Select **Create webhook**.
+
+To change the callback URL, description, and payment event subscriptions select **Manage webhook** from the **Webhooks** page in the GOV.UK Pay admin tool.
+
+### Deactivate a webhook
+
+You can deactivate a webhook at any time. Deactivated webhooks do not send events to your callback URL.
+
+To deactivate your webhook:
+
+1. Open the **Webhooks** page in the GOV.UK Pay admin tool.
+
+1. Select **Manage webhook**.
+
+1. Select **Deactivate webhook**.
+
+You can reactivate the webhook at any time.
+
+## Receiving webhook messages
+
+Once you’ve created a webhook, GOV.UK Pay sends a webhook message to your callback URL after any of the events you subscribed to.
+
+Your service must send a `2xx` successful response within 15 seconds.
+
+<%= warning_text('Your service may receive webhooks messages in an unexpected order. Make sure that your integration does not depend on receiving events in a particular order.') %>
+
+### Verifying webhook messages from GOV.UK Pay
+
+Every webhook message from GOV.UK Pay is signed with the `Pay-Signature` header. The value of this header is a lower-case, hexadecimal hash-based message authentication code (HMAC) of the webhook message body, generated using the SHA-256 hash function. Your webhook signing secret is used as the key.
+
+To view your webhook’s signing secret, select **Manage signing secret** from the **Manage webhook** page in the [GOV.UK Pay admin tool](https://selfservice.payments.service.gov.uk/login).
+
+Use your signing secret and the webhook message payload to verify the webhook message:
+
+1\. Generate a HMAC of the webhook message body. When generating the HMAC:
+
+* use the SHA-256 hash function
+* the message is the webhook message body, which is encoded as UTF-8 with no byte order mark (BOM)
```

Comment:
Thinking about it “HTTP request body” might be clearer than “webhook message body”.